src/main/java/kostka/moviecatalog/service/DbMovieService.java

```diff
@@ -63,15 +64,22 @@ public Movie getMovie(final Long movieId) {
         return movieRepository.findByIdInOrderByIdDesc(ids);
     }
 
-    public List<Movie> get5LatestMovies() {
+    public List<Movie> get5LatestMoviesFromCache() {
         List<Long> longIds = getMovieIdsFromRedisCache(LATEST_MOVIES_KEY);
         return movieRepository.findByIdInOrderByIdDesc(longIds);
     }
 
     public List<Movie> getTop5RatingMoviesFromDB() {
-        return movieRepository.findTop5ByOrderByRatingDesc();
+        List<Movie> movies = movieRepository.findTop5ByOrderByRatingDesc();
+        return movies.isEmpty() ? new ArrayList<>() : movies;
     }
 
+    public List<Movie> get5LatestMoviesFromDB() {
+        List<Movie> movies = movieRepository.findTop5ByOrderByIdDesc();
```

Comment:
pokud by bylo nutne to, na co se ptal Radek, tak pak je tady duplicita, ktera by sla extrahovat do funkce `getList` nebo tak neco.

Reply:
Radkuv koment byl opodstatneny :)